opentelemetry-resourcedetector-gcp/src/opentelemetry/resourcedetector/gcp_resource_detector/_metadata.py

```diff
@@ -34,13 +35,14 @@ class Project(TypedDict):
 
 
 Attributes = TypedDict(
-    "Attributes", {"cluster-location": str, "cluster-name": str}
+    "Attributes", {"cluster-location": str, "cluster-name": str}, total=False
 )
 
 
 class Instance(TypedDict):
     attributes: Attributes
-    id: str
+    # id can be an integer on GCE VMs or a string on other environments
```

Comment:
@dashpole updated, does this address your comment?

Reply:
yep